snowflake/pkg/snowflake/dsn.go

```diff
@@ -47,7 +49,7 @@ func GetDSN(options ...func(*sf.Config)) (string, *sf.Config, error) {
 			return value
 		}
 		if failOnMissing {
-			log.Fatalf("%v environment variable is not set.", k)
+			logFatalf("%v environment variable is not set.", k)
```

Comment:
I think we should just change that logic and have the `GetDSN` function return an error instead of calling `log.Fatal`